src/lib/php/Dao/ClearingDao.php

```diff
@@ -756,41 +756,73 @@ public function markDirectoryAsIrrelevant(ItemTreeBounds $itemTreeBounds,$groupI
    * @param int $groupId
    * @param int $userId
    */
-  protected function markDirectoryAsIrrelevantIfScannerDetected(ItemTreeBounds $itemTreeBounds,$groupId,$userId)
+  public function deleteIrrelevantDecisionsFromDirectory(ItemTreeBounds $itemTreeBounds,$groupId,$userId)
   {
-    $params = array($itemTreeBounds->getLeft(), $itemTreeBounds->getRight(), $userId, $groupId, DecisionTypes::IRRELEVANT, DecisionScopes::ITEM);
+    $this->markDirectoryAsIrrelevantIfScannerDetected($itemTreeBounds, $groupId, $userId, true);
+    $this->markDirectoryAsIrrelevantIfUserEdited($itemTreeBounds, $groupId, $userId, true);
+  }
+  
+  /**
+   * @param ItemTreeBounds $itemTreeBounds
+   * @param int $groupId
+   * @param int $userId
+   */
+  protected function markDirectoryAsIrrelevantIfScannerDetected(ItemTreeBounds $itemTreeBounds,$groupId,$userId,$removeDecision=false)
+  {
+    $statementName = __METHOD__ ;
+    $params = array($itemTreeBounds->getLeft(), $itemTreeBounds->getRight());
+    $params[] = $groupId;
+    $a = count($params);
     $options = array(UploadTreeProxy::OPT_SKIP_THESE=>'noLicense',
                      UploadTreeProxy::OPT_ITEM_FILTER=>' AND (lft BETWEEN $1 AND $2)',
-                     UploadTreeProxy::OPT_GROUP_ID=>'$4');
+                     UploadTreeProxy::OPT_GROUP_ID=>'$'.$a.'');
     $uploadTreeProxy = new UploadTreeProxy($itemTreeBounds->getUploadId(), $options, $itemTreeBounds->getUploadTreeTableName());
-    $statementName = __METHOD__ ;
-    $sql = $uploadTreeProxy->asCte()
-        .' INSERT INTO clearing_decision (uploadtree_fk,pfile_fk,user_fk,group_fk,decision_type,scope) 
-          SELECT uploadtree_pk itemid,pfile_fk pfile_id, $3, $4, $5, $6 FROM UploadTreeView';
+    if(!$removeDecision){
+      $params[] = $userId;
+      $params[] = DecisionTypes::IRRELEVANT;
+      $params[] = DecisionScopes::ITEM;
+      $sql = $uploadTreeProxy->asCte()
+          .' INSERT INTO clearing_decision (uploadtree_fk,pfile_fk,user_fk,group_fk,decision_type,scope) 
+            SELECT uploadtree_pk itemid,pfile_fk pfile_id, $'.($a+1).', $'.$a.', $'.($a+2).', $'.($a+3).' FROM UploadTreeView
+	    WHERE NOT EXISTS (SELECT uploadtree_fk FROM clearing_decision WHERE decision_type=$5 and uploadtree_fk=UploadTreeView.uploadtree_pk)';
+    }else{
+      $params[] = DecisionTypes::IRRELEVANT;
+      $sql = $uploadTreeProxy->asCte()
+          .' DELETE FROM clearing_decision WHERE clearing_decision_pk IN (SELECT clearing_decision_pk FROM clearing_decision WHERE 
+           decision_type = $'.($a+1).' AND uploadtree_fk IN (SELECT uploadtree_pk FROM UploadTreeView) ORDER BY clearing_decision.date_added DESC)';
```

Comment:
Is the `ORDER BY` statement really necessary?

Reply:
Yeah, its required to get the recent record.

Reply:
Sorry, but I don't really get it.  Most probably, I have an error in my thinking, but doesn't `$sql` essentially look like
`[CTE] DELETE FROM table WHERE field in ( *something* ORDER BY date DESC) `  ?
Why does the ordering make a difference here?